packages/rx_bloc_cli/mason_templates/bricks/lib_auth_apple/__brick__/lib/lib_auth_apple/services/apple_login_service.dart

```diff
@@ -0,0 +1,42 @@
+{{> licence.dart }}
+
+import 'package:sign_in_with_apple/sign_in_with_apple.dart';
+
+import '../../assets.dart';
+import '../../base/models/errors/error_model.dart';
+import '../../lib_auth/models/auth_token_model.dart';
+import '../../lib_auth/services/social_login_service.dart';
+import '../models/apple_credentials_model.dart';
+import '../repositories/apple_auth_repository.dart';
+
+class AppleLoginService extends SocialLoginService {
+  AppleLoginService(this._appleAuthRepository);
+
+  final AppleAuthRepository _appleAuthRepository;
+
+  @override
+  Future<AuthTokenModel> getSocialAuthentication() async {
+    final AuthorizationCredentialAppleID credentials =
+        await SignInWithApple.getAppleIDCredential(
```

Comment:
Tight coupling of the `sign_in_with_apple` makes the service not testable. Please move the SignInWithApple interaction to the data layer

Reply:
A new DataSource is added.